### PR TITLE
Feat/translation fr

### PR DESCRIPTION
**Pull request by matcho**
Hi.

Here is a french translation for the plugin.

Greetings,
Mathias

### CODE REVIEW

File: footer.php [hunk 1]
```diff
@@ -185,17 +172,31 @@ function draw_navbuttons() {
...
+            $home->link = new moodle_url('/course/view.php', array('id'=>$COURSE->id));
+            $home->name = get_string('coursepage','block_navbuttons');
+        } else {
+            $home->link = $CFG->wwwroot;
+            $home->name = get_string('frontpage','block_navbuttons');
+        }
+        list($icon, $bgcolour) = navbutton_get_icon($settings->buttonstype, 'home', $context, BLOCK_NAVBUTTONS_HOMEICON, $settings->backgroundcolour, $settings->customusebackground);
+        $output .= make_navbutton($icon, $bgcolour, $home->name, $home->link, false, "home");
+    }
+
```
Comment (davosmith) [inline footer.php]: I'm not convinced that moving the 'home' button down here is an improvement - I prefer it at the start.

File: footer.php [hunk 2]
```diff
@@ -185,17 +172,31 @@ function draw_navbuttons() {
         }
         if ($first) {
             list($icon, $bgcolour) = navbutton_get_icon($settings->buttonstype, 'first', $context, BLOCK_NAVBUTTONS_FIRSTICON, $settings->backgroundcolour, $settings->customusebackground);
-            $output .= make_navbutton($icon, $bgcolour, $first->name, $first->link);
+            $output .= make_navbutton($icon, $bgcolour, $first->name, $first->link, false, "first");
```
Comment (davosmith) [inline footer.php]: If you're going to add a new param (and this seems like a good idea to me), please put it before the optional final param (as it doesn't make sense to have to specify 'false' on every call, just to allow you to then specify the class name).

File: footer.php [hunk 3]
```diff
@@ -282,6 +283,10 @@ function make_navbutton($imgsrc, $bgcolour, $title, $url, $newwindow = false) {
         }
         $output .= 'margin-right: 5px;" width="50" height="50" /></a>';
     } else {
+		$formClass = 'navbuttontext';
+		if ($classes != "") {
+		    $formClass .= " " . $classes;
+		}
```
Comment (davosmith) [inline footer.php]: Please indent with spaces, not tabs.

Also, the comparison does not make sense - the default for the $classes parameter is null, but you then check it against "" to see if it has been specified. Far better to just write 'if ($classes) {' instead, as that is more consistent.

File: footer.php [hunk 4]
```diff
@@ -282,6 +283,10 @@ function make_navbutton($imgsrc, $bgcolour, $title, $url, $newwindow = false) {
         }
         $output .= 'margin-right: 5px;" width="50" height="50" /></a>';
     } else {
+		$formClass = 'navbuttontext';
```
Comment (davosmith) [inline footer.php]: Please don't use camelcase for variables - $formclass is the correct name, according to the Moodle coding guidelines.

File: footer.php [hunk 5]
```diff
@@ -93,7 +93,7 @@ function draw_navbuttons() {
         if ($mod->sectionnum > 0 && $sectionnum != $mod->sectionnum) {
```
Comment (davosmith) [inline footer.php]: Please could the commit message be in English, so I know what it is doing?

File: footer.php [hunk 6]
```diff
@@ -93,7 +93,7 @@ function draw_navbuttons() {
...
             $thissection = $sections[$mod->sectionnum];
 
-            if ($thissection->visible || !$COURSE->hiddensections ||
+            if ($thissection->visible || /*!$COURSE->hiddensections ||*/
```
Comment (davosmith) [inline footer.php]: If you're removing obsolete code, please delete it, rather than commenting it out.

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1 @@
+/nbproject
```
Comment (davosmith) [inline .gitignore]: I don't think I need to add an IDE-specific .gitignore file - if you are having a local problem with netbeans generating unwanted files, please add them to .git/info/exclude (which is not checked in as part of the repo).

File: lang/fr/block_navbuttons.php [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
```
Comment (davosmith) [inline lang/fr/block_navbuttons.php]: This all looks OK to me, but I can't merge this without also pulling in all the other changes in the branch (some of which are unwanted, some of which need fixing).

**Comment by davosmith** [comment]
Thanks for this - please can you see the comments I've added to the individual commits and address the issues I've highlighted before I can merge the changes.

Davo

**Comment by matcho** [comment]
Hi,

Mhh it seems I've done all the errors I could, here :)
I'm such a beginner with Git... I'll try to fix this today.
Note : the pull request for translation was intended to be separate from the other changes, but I messed with the branch...

Mathias

**Comment by davosmith** [comment]
Not a problem - I look forward to seeing the final version.

**Comment by matcho** [comment]
I tried to modify the code according to your comments.
This led to 2 drawbacks :
- 2 revert commits
- the branch name does not reflect all changes

I can delete the branch and make a proper new one if you prefer

**Comment by davosmith** [comment]
I'd recommend using 'git rebase -i' to merge/rewrite the scripts, then 'git push --force' to overwrite the existing branch.

If that's a bit too much, then creating a fresh branch may be easier to manage.

**Pull request closed without merging**